mongodb-dbpath-mount.md (#85)
percona-platform/portal-doc: docs/advisors/checks/mongodb-dbpath-mount.md

# MongoDB DBPATH mount

## Description

> This check returns a warning if dbpath does not have a dedicated mount point.

Reviewer: ```suggestion
This check warns if **dbpath** does not have a dedicated mount point.
```

## Resolution

> For tuning flexibility, we recommended that MongoDB data sits on its own disk volume, preferably with its own dedicated disks/RAID array. While it may complicate backups, for the best performance you can also dedicate a separate volume for the MongoDB journal to separate its disk activity noise from the main data set. The journal does not yet have its own config/command-line setting, so you’ll need to mount a volume to the “journal” directory inside the dbPath. For example, “/var/lib/mongo/journal” would be the journal mount-path if the dbPath was set to “/var/lib/mongo”.

Reviewer: ```suggestion
For tuning flexibility, Percona recommends that MongoDB data sit on its own disk volume, preferably with its own dedicated disks/RAID array. Since this may complicate backups, for best performance you can also dedicate a separate volume for the MongoDB journal. This will separate its disk activity noise from the main data set. 

The journal does not have its own config/command-line setting yet. Therefore, you’ll need to mount a volume to the “journal” directory inside the dbPath. For example, “_/var/lib/mongo/journal_” would be the journal mount-path if the dbPath was set to “_/var/lib/mongo_”.
```

> There are several reasons why it is recommended to use a separate mount point for the dbPath in MongoDB:

Reviewer: ```suggestion
Using a separate mount point for the **dbPath** in MongoDB is recommended for the following reasons:
```

- **Performance:** Separating the data files from the system files can help improve the performance of MongoDB. This is because the data files can be placed on a dedicated disk or disk partition, which can improve read and write performance.

> - **Scalability:** By separating the data files from the system files, it becomes easier to scale the database. You can add more storage as needed without having to worry about running out of space on the system disk.

Reviewer: ```suggestion
- **Scalability:** By separating the data files from the system files, scaling the database becomes easier. You can add more storage as needed without having to worry about running out of space on the system disk.
```
